text/0000-pub-sub.md

```diff
@@ -0,0 +1,170 @@
+
+#### Registration
+
+Each module which acts as a publisher or a subscriber must contain a configuration or property with a set of event types it deals with. Publishers should provide event descriptors. For subscribers, it is enough to specify event types together with endpoints for event delivery. Details of that configuration must be defined further.
+
+##### Event Type registration 
+The set of event descriptors will be registered by a publisher at the time when the module is enabled for a tenant. To do that the publisher must call `pubsub` API for event type registration. The payload for that call is a set of event descriptors which define events this particular module is able to publish.
+
+At this stage it is possible to check if another module has already been registered as a publisher for the same event descriptors.
+
+It is worth mentioning that at this stage only event descriptors will be registered. To allow a module to publish events further steps optionally should be taken.
+
+![Event type registration diagram](http://www.plantuml.com/plantuml/png/TP4nJyCm48Lt_ufJ9nWgTKPL9nZO4B01myM-D2SSsxBF5VttN1nQXL8YPOZFxxltNeJ54pmSA0Z8CL61J7ikab4u1cEmIWnrx2Z3QjljtZQRpSVAc8HPs792bQt6wDRRL3lFPQ2yMm4MICROw3tP2LFcEtgbrAwDoP4nwnsaEq3GMLKJhCS1Eq_kFVGnwq3qEuWn-nnhMfZyxH7qIJ33dWcT2Wi9n1weAHmLcpb9p4N1EOvwmSRGyxvU4jJMn9ZW6dXU-DfQkVaFUTlKAZphdp0NkhORWmCxaFs45izdDNlyoKAPt79Xew8Nu1uSzxtETVG7saWo41GSJpjRG_Ibq0PPsDE27iC1FPzMtPz9SmglnaDixNYsCk5hS9UObl95HLTImXJIS_W4YStO_7ElE5csE-21e2CAYpPcHxfzUhy1)
+
+##### Subscriber registration
+A back-end module can register event types it can consume at the time when the module is enabled for a tenant. To do that it must call the `pubsub` API for registration. The payload for that call is a set of event types augmented with endpoints provided by this module. These endpoints will be used to deliver events of respective types.
+
+##### Publisher registration
+When a module is up and running and is enabled for a tenant, a tenant administrator can define types of events which must (or must not) be published by this module in scope of this tenant (subscribe a tenant to event types) using either UI Admin console/Settings Page or HTTP requests to `pubsub` endpoint for bulk operations. 
+
+It is suggested that by default all event types are activated for a tenant.
+
+##### Cancellation of registration
+When a module is disabled for a tenant whether it is a publisher or a subscriber the module must unregister event types it deals with by calling the API provided by `pubsub`.
+
+##### mod-pubsub responsibilities
+The `mod-pubsub` must keep track of registered event types, publishers and subscribers which are able to send and receive events of those types.
+
+
+#### Data structures
+##### Event descriptor data structure
+Name | Type
+--- | ---
+Event type | String
```

Comment:
This is the unique identifier for an event type, right?  But it's not a UUID, but rather a human-readable string?

Reply:
Yes you are right